connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/outbound/lifecycle/OutboundConnectorManager.java

```diff
@@ -101,4 +104,20 @@ private void openWorkerForOutboundConnector(
 
     jobWorkerManager.openWorker(client, zeebeWorkerValue, connectorJobHandler);
   }
+
+  /**
+   * Adds the retry context input variable to the list of input variables, if it is not already.
+   * This is required so that we can retrieve the retry context (attempts by errorCode) from the job
+   * when throwing a {@link ConnectorRetryException}. This method is idempotent, meaning that it
+   * will not add the retry context input variable if it is already present in the list of input
+   * variables.
+   *
+   * @param inputVariables the list of input variables
+   * @return the array of input variables including the retry context input variable
+   */
+  private String[] addRetryContextInputVariable(String[] inputVariables) {
+    var variableSet = new HashSet<>(Arrays.asList(inputVariables));
+    variableSet.add(ConnectorRetryException.RETRY_CONTEXT_INPUT_VARIABLE);
+    return variableSet.toArray(new String[0]);
+  }
```

Comment:
Yeah, checked that. So it's identical to `toArray()`, since a new array will be allocated ([ref](https://docs.oracle.com/javase/8/docs/api/java/util/Set.html#toArray-T:A-)).
I think, just `toArray()` should be more efficient.

Reply:
but `toArray()` can't return a typed array it returns an `Object[]`

Reply:
aaahhh 🤯 
Thanks!